modules/swagger-codegen/src/main/resources/Java/ApiClient.mustache

```diff
@@ -421,10 +421,10 @@ public class ApiClient {
       throw new ApiException(500, "unknown method type " + method);
     }
 
-    if(response.getClientResponseStatus() == ClientResponse.Status.NO_CONTENT) {
+    if(response.getStatusInfo() == ClientResponse.Status.NO_CONTENT) {
```

Comment:
https://github.com/swagger-api/swagger-codegen/pull/3963

The integration tests from Maven do not cover this case. I tried setting up rails-petstore and tried testing against it, but rails-petstore does not have a response that covers the pull request.